lib/vm.js

```diff
@@ -400,6 +403,45 @@ const vmConstants = {
 
 ObjectFreeze(vmConstants);
 
+function stripTypeScriptTypes(code, options = {}) {
+  emitExperimentalWarning('vm.stripTypeScriptTypes');
+  const { mode = 'strip-only', sourceMap = false, filename = '' } = options;
+  validateObject(options, 'options');
```

Comment:
The order seems wrong. if `options` was provided with any non-object value other than `undefined` we should throw an `ERR_INVALID_ARG_TYPE`, but this currently could potentially throw an error such as `can't destructure property of options as it is null`

```suggestion
  validateObject(options, 'options');
  const { mode = 'strip-only', sourceMap = false, filename = '' } = options;
```